controllers/dbaasprovider_controller.go

```diff
@@ -110,200 +111,3277 @@ func (r *DBaaSProviderReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// RDS controller registration custom resource isn't present,so create now with ClusterRole owner for GC
+	opts := &client.ListOptions{
+		LabelSelector: label.SelectorFromSet(map[string]string{
+			"olm.owner":      r.operatorNameVersion,
+			"olm.owner.kind": "ClusterServiceVersion",
+		}),
+	}
+	clusterRoleList := &rbac.ClusterRoleList{}
+	if err := r.List(context.Background(), clusterRoleList, opts); err != nil {
+		logger.Error(err, "unable to list ClusterRoles to seek potential operand owners")
+		return ctrl.Result{}, err
+	}
+
+	if len(clusterRoleList.Items) < 1 {
+		err := errors.NewNotFound(
+			schema.GroupResource{Group: "rbac.authorization.k8s.io", Resource: "ClusterRole"}, "potentialOwner")
+		logger.Error(err, "could not find ClusterRole owned by CSV to inherit operand")
+		return ctrl.Result{}, err
+	}
+
```

Comment:
Not saying it's wrong, just checking - It's ok to lose the 'if' check wrapping this? We don't mind always performing this bit when we pass through here?

Reply:
Do you mean `if len(clusterRoleList.Items) < 1`?
The code needs one `clusterrole` to [generate](https://github.com/RHEcosystemAppEng/rds-dbaas-operator/blob/7d573c4ec5e0ac32fa7d9f1861c52a1ef0f72f6f/controllers/dbaasprovider_controller.go#L159-L160) the dbaasprovider CR.
This solution is not designed by me orginally, other provider [operators](https://github.com/CrunchyData/crunchy-bridge-operator/blob/e55bad94135a1293e2316c7eb1656b2ac31d4b08/controllers/dbaas.redhat.com/dbaasprovider_reconciler.go#L174-L175) works like this as well.

Reply:
I meant where this full block was previously wrapped by a Get check and only followed these steps when NotFound, however, I pulled the full file to compare and see where the predicate means we should encounter that scenario in the controller.

Reply:
The code is changed to do a `CreateOrUpdate`. So if the dbaasprovider object exists, the code will update it, which is necessary when we need to change the dbaasprovider object for upgrade.